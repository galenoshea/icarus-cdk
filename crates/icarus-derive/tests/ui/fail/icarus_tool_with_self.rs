use icarus_derive::*;

struct TestStruct;

impl TestStruct {
    #[icarus_tool("Tool with self parameter")]
    fn bad_tool(&self) -> Result<String, String> {
        Ok("Should fail".to_string())
    }
}

fn main() {}