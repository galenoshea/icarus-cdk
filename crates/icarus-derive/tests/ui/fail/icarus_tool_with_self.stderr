error: Tool function validation failed:
         - Tool functions must have either #[query] or #[update] attribute
         - Tool functions cannot have self parameters
 --> tests/ui/fail/icarus_tool_with_self.rs:7:8
  |
7 |     fn bad_tool(&self) -> Result<String, String> {
  |        ^^^^^^^^
